File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) 
    {
       int nonDuplicateIndex = 0;
       int lastNonDuplicate = nums[0];
       int i = 1;

       while (i < nums.Length && nonDuplicateIndex < nums.Length)
       {
            while(nonDuplicateIndex < nums.Length && nums[nonDuplicateIndex] == lastNonDuplicate)
                nonDuplicateIndex++;
            
            if(nonDuplicateIndex < nums.Length)
            {
                nums[i] = nums[nonDuplicateIndex];
                lastNonDuplicate = nums[i];
                i++;
            }
            
       }
        return i;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        nums.sort()
        output = []
        self.backtrack(nums, target, 0, [], output)
        return output

    def backtrack(self, nums: List[int], target: int,
     index: int, current: List[int], output: List[List[int]]):
        
        cur_list_sum = sum(current)

        if cur_list_sum == target:
            output.append(list(current))
            return
        
        if index >= len(nums) or cur_list_sum > target:
            return
        else:
            current.append(nums[index])
            self.backtrack(nums, target, index, current, output)

            # Backtrack
            current.pop()
            self.backtrack(nums, target, index + 1, current, output)